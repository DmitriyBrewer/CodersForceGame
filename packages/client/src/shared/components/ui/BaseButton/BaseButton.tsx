import { Button } from '@mui/material'

const BaseButton = Button

export default BaseButton
