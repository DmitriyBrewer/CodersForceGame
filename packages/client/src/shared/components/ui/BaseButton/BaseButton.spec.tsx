import { render, screen } from '@testing-library/react'
import { act } from 'react'

import '@testing-library/jest-dom'
import BaseButton from './BaseButton'

describe('<BaseButton />', () => {
  const handleClick = jest.fn()

  it('Snapshot is correct', () => {
    let asFragment: () => DocumentFragment = () => document.createDocumentFragment()

    act(() => {
      const result = render(<BaseButton onClick={handleClick} />)
      asFragment = result.asFragment
    })
    expect(asFragment()).toMatchSnapshot()
  })

  it('Renders correctly with default props', () => {
    act(() => {
      render(<BaseButton />)
    })
    const button = screen.getByRole('button')
    expect(button).toMatchSnapshot()
  })
})
