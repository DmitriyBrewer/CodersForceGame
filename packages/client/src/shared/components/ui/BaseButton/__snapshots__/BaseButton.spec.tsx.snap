// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseButton /> Renders correctly with default props 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`<BaseButton /> Snapshot is correct 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
